src/comotion/dash.py

```diff
@@ -506,109 +1258,157 @@ def create_gzipped_csv_stream_from_df(df: pd.DataFrame) -> io.BytesIO:
-    dash_table: str
-        name of Dash table to upload the file to
-    dash_orgname: str
-        orgname of your Dash instance
-    dash_api_key: str
-        valid api key for Dash API
-    encoding: str
-        the encoding of the source file.  defaults to utf-8.
-    chunksize: int
-        (optional)
-        the maximum number of lines to be included in each file.
-        Note that this should be low enough that the zipped file is less
-        than Dash maximum gzipped file size. Defaults to 30000.
-    modify_lambda:
-        (optional)
-        a callable that recieves the pandas dataframe read from the
-        csv.  Gives the opportunity to modify - such as adding a timestamp
-        column.
-        Is not required.
-    path_to_output_for_dryrun: str
-        (optional)
-        if specified, no upload will be made to dash, but files
-        will be saved to the location specified. This is useful for
-        testing.
-        multiple files will be created: [table_name].[i].csv.gz where i
-        represents multiple file parts
-    service_client_id: str
-        (optional)
-        if specified, specifies the service client for the upload. See the dash documentation for an explanation of service client.
+        Either a path to the file to be uploaded, or a FileIO stream representing the file to be uploaded.
+        Should be an unencrypted, uncompressed CSV file.
+    dash_table : str
+        Name of the Dash table to upload the file to.
+    dash_orgname : str
+        Orgname of your Dash instance.
+    file_key : str, optional
+        A unique key for the file being uploaded. If not provided, a key will be generated.
+    dash_api_key : str, optional
+        Valid API key for Dash API. Required for v1 data model uploads.
+    check_sum : Optional[Dict[str, Union[int, float, str]]], optional
+        Checksum data for the files to be committed. The checksum should be a dictionary
+        with Presto/Trino expressions as keys and their expected results as values.
+    encoding : str, default 'utf-8'
+        The encoding of the source file.
+    chunksize : int, default 30000
+        The maximum number of lines to be included in each file. Note that this should be low enough
+        that the zipped file is less than Dash's maximum gzipped file size.
+    modify_lambda : Callable, optional
+        A callable that receives the pandas DataFrame read from the CSV. Provides the opportunity to modify
+        the DataFrame, such as adding a timestamp column.
+    path_to_output_for_dryrun : str, optional
+        If specified, no upload will be made to Dash, but files will be saved to the location specified.
+        This is useful for testing. Multiple files will be created (1 per chunk)
+    service_client_id : str, optional
+        If specified, specifies the service client for the upload. See the Dash documentation for an explanation
+        of the service client.
```

Comment:
we can include a link to a page - we must have a proper explanation in the docs on service client and how it works with loads.